lib/linguist/languages.yml

```diff
@@ -1758,6 +1758,16 @@ Groovy Server Pages:
   codemirror_mode: htmlembedded
   codemirror_mime_type: application/x-jsp
   language_id: 143
+HAProxy:
+  type: data
+  extensions:
+  - ".cfg"
```

Comment:
If the filename of these files is known, do we need to add the extension as well?

Reply:
The file will almost always be called haproxy.cfg, but there's nothing stopping someone from naming it something else. A person could split their configuration into multiple files line haproxy-1.cfg or sample.cfg.

Reply:
If we keep the `.cfg` extension will need a search query to estimate the usage of that extension for HAProxy configuration files on github.com.

Reply:
I updated the PR comment with a second search query for files that contain "roundrobin" that have the .cfg extension and are _not_ haproxy.cfg.